File: Java Fundamentals/Data types and variables/Exercise Tasks/Task11_Snowballs.java
import java.util.Scanner;
//100/100
public class Task11_Snowballs {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int N = Integer.parseInt(scanner.nextLine());
        double highestSnowballValue = 0;
        double highestSnowballSnow = 0;
        double highestSnowballTime = 0;
        double highestSnowballQuality = 0;
        long snowballValue = 0;

        for (int i = 0; i < N; i++) {
            int snowballSnow = Integer.parseInt(scanner.nextLine());
            int snowballTime = Integer.parseInt(scanner.nextLine());
            int snowballQuality = Integer.parseInt(scanner.nextLine());

            snowballValue = (long) Math.pow((snowballSnow/snowballTime), snowballQuality);

            if (snowballValue > highestSnowballValue) {
                highestSnowballValue = snowballValue;
                highestSnowballSnow = snowballSnow;
                highestSnowballQuality = snowballQuality;
                highestSnowballTime = snowballTime;
            }

        }
        System.out.printf("%.0f : %.0f = %.0f (%.0f)", highestSnowballSnow, highestSnowballTime, highestSnowballValue, highestSnowballQuality);
    }
}

File: C#/C# Fundamentals/Data Type & Variables/Exercise Tasks/Task11_Snowballs.cs
using System;

namespace Task11_Snowballs
{
    class Program
    {
        static void Main(string[] args)
        {
            int numberOfSnowballs = int.Parse(Console.ReadLine());
          
            long snowballValue = 0;
            long maxValue = 0;
            int maxSnow = 0;
            int maxTime = 0;
            int maxQuality = 0;

            for (int i = 0; i < numberOfSnowballs; i++)
            {
                int snowballSnow = int.Parse(Console.ReadLine());
                int snowballTime = int.Parse(Console.ReadLine());
                int snowballQuality = int.Parse(Console.ReadLine());

                snowballValue = (long) Math.Pow(snowballSnow / snowballTime, snowballQuality);
            if(snowballValue > maxValue)
                {
                    maxValue = snowballValue;
                    maxSnow = snowballSnow;
                    maxTime = snowballTime;
                    maxQuality = snowballQuality;
                }
            }

            Console.WriteLine($"{maxSnow} : {maxTime} = {maxValue} ({maxQuality})");
        }
    }
}
